nova/src/provider/pedersen.rs

```diff
@@ -1,30 +1,51 @@
 use std::marker::PhantomData;
 
 use crate::{commitment::CommitmentScheme, LOG_TARGET};
-use ark_ec::{CurveGroup, ScalarMul, VariableBaseMSM};
-use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_ec::{
+    short_weierstrass::{Affine, Projective, SWCurveConfig},
+    ScalarMul, VariableBaseMSM,
+};
+use ark_ff::{field_hashers::hash_to_field, PrimeField};
+use sha3::digest::{ExtendableOutput, Update};
+
+pub use crate::provider::hashtocurve::SVDWMap;
 
 #[derive(Debug)]
 pub struct PedersenCommitment<G>(PhantomData<G>);
 
-impl<G> CommitmentScheme<G> for PedersenCommitment<G>
+impl<G> CommitmentScheme<Projective<G>> for PedersenCommitment<G>
 where
-    G: CurveGroup,
-    G::MulBase: CanonicalSerialize + CanonicalDeserialize,
+    G: SWCurveConfig + SVDWMap,
+    G::BaseField: PrimeField,
 {
-    type PP = Vec<G::MulBase>;
-    type SetupAux = ();
+    type PP = Vec<Affine<G>>;
+    type SetupAux = [u8];
 
-    type Commitment = G;
+    type Commitment = Projective<G>;
 
-    fn setup(n: usize, _aux: &Self::SetupAux) -> Self::PP {
-        // TODO: replace test rng.
-        let mut rng = ark_std::test_rng();
-        let ps: Vec<G> = (0..n).map(|_| G::rand(&mut rng)).collect();
-        ScalarMul::batch_convert_to_mul_base(&ps)
+    fn setup(n: usize, _bytes: &Self::SetupAux) -> Self::PP {
+        let _span = tracing::debug_span!(
+            target: LOG_TARGET,
+            "pedersen::setup",
+            %n,
+        )
+        .entered();
+
+        #[cfg(not(test))]
+        let bases = batch_map_to_curve(b"from_uniform_bytes", _bytes, n);
```

Comment:
Is this domain separator standardized anywhere? It seems to me like it might make more sense to describe the functionality being implemented along with the encoding, as the encoding might be the same in other domains (i.e., use `setup_pedersen_provider_from_uniform_bytes` or similar).

Reply:
I copied it from Microsoft Nova

Reply:
Admittedly it's unlikely we'll see a conflict, but given that we're much more likely to build other functionality that will involve hashing to curves at some point than their protocol reference implementation I think being a bit more explicit would be preferable.